src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/ReadOnlyTensorSpan.cs

```diff
@@ -527,8 +526,8 @@ public void CopyTo(scoped TensorSpan<T> destination)
                 if (Rank > TensorShape.MaxInlineRank)
                 {
                     curIndexesArray = ArrayPool<nint>.Shared.Rent(Rank);
-                    curIndexes = curIndexesArray;
-                    curIndexes = curIndexes.Slice(0, Rank);
+                    curIndexes = curIndexesArray.AsSpan(0, Rank);
+                    curIndexes.Clear();
```

Comment:
Should this be moved down?

There's no guarantee that `stackalloc` is zeroed and if we ever add `SkipLocalsInit` (a lot of in-box code has this), then it may not be.

Reply:
If there is no guarantee that `stackalloc` is zeroed then yeah it should be moved down.